File: scmp241/py2cpp/examples/06_monotonicity.cpp
#include <iostream>
#include <vector>
#include <ranges>

bool is_strictly_increasing(const std::vector<int>& sequence)
{
    for (auto i : std::views::iota(1u, sequence.size())) {
        if (sequence.at(i) <= sequence.at(i-1)) {
            return false;
        }
    }
    return true;
}

int main()
{
    const std::vector primes = {2, 3, 5, 7, 11, 13, 17, 19, 23, 29};
    std::cout << "Prime sequence is ";
    if (!is_strictly_increasing(primes)) std::cout << "not ";
    std::cout << "strictly increasing.\n";

    const std::vector fibonacci = {0, 1, 1, 2, 3, 5, 8, 13, 21, 34};
    std::cout << "Fibonacci sequence is ";
    if (!is_strictly_increasing(fibonacci)) std::cout << "not ";
    std::cout << "strictly increasing.\n";
}

File: scmp241/py2cpp/examples/06_monotonicity.py
def is_strictly_increasing(sequence: list) -> bool:
    for i in range(1, len(sequence)):
        if sequence[i] <= sequence[i-1]:
            return False
    return True

if __name__ == '__main__':
    primes = [2, 3, 5, 7, 11, 13, 17, 19, 23, 29]
    print('Prime sequence is ', end='')
    if not is_strictly_increasing(primes): print('not ', end='')
    print('strictly increasing.')

    fibonacci = [0, 1, 1, 2, 3, 5, 8, 13, 21, 34]
    print('Fibonacci sequence is ', end='')
    if not is_strictly_increasing(fibonacci): print('not ', end='')
    print('strictly increasing.')